### PR TITLE
Migrate to Go Modules

### PR DESCRIPTION
**Pull request by mrutkows**
Update to Go 1.15, remove deprecated godeps/vendor in favor of go.mod (go mod).

### CODE REVIEW

**Comment by mrutkows** [comment]
@rabbah @jiangpengcheng  Please review if able, thanks!

File: README.md [hunk 1]
```diff
@@ -21,39 +21,76 @@
...
-- [govendor](https://github.com/kardianos/govendor)
+The Openwhisk Go Client library requires you to [Download and install GoLang](https://golang.org/dl/) onto your local machine.
+
+> **Note** Go version 1.15 or higher is recommended
 
 Make sure you select the package that fits your local environment, and [set the GOPATH environment variable](https://github.com/golang/go/wiki/SettingGOPATH).
 
+### GoLang setup
+
+The wskdeploy utility is a GoLang program so you will first need to [Download and install GoLang](https://golang.org/dl/) onto your local machine.
```
Comment (rabbah) [inline README.md:40]: why is this mentioned here?

File: README.md [hunk 2]
```diff
@@ -21,39 +21,76 @@
...
+
+> **Note** Go version 1.15 or higher is recommended
+
+Make sure your `$GOPATH` is defined correctly in your environment. For detailed setup of your GoLang development environment, please read [How to Write Go Code](https://golang.org/doc/code.html).
+
+### Download the source code from GitHub
+
+As the code is managed using GitHub, it is easiest to retrieve the code using the `git clone` command.
+
+if you just want to build the code and do not intend to be a Contributor, you can clone the latest code from the Apache repository:
```
Comment (rabbah) [inline README.md:50]:
```suggestion
If you just want to build the code and do not intend to be a Contributor, you can clone the latest code from the Apache repository:
```

File: README.md [hunk 3]
```diff
@@ -21,39 +21,76 @@
...
+
+```sh
+git clone git@github.com:apache/openwhisk-client-go
+```
+
+or you can specify a release (tag) if you do not want the latest code by using the `--branch <tag>` flag. For example, you can clone the source code for the tagged 1.1.0 [release](https://github.com/apache/openwhisk-client-go/releases)
```
Comment (rabbah) [inline README.md:56]:
```suggestion
You can also specify a release (tag) if you do not want the latest code by using the `--branch <tag>` flag. For example, you can clone the source code for the tagged 1.1.0 [release](https://github.com/apache/openwhisk-client-go/releases)
```

**Review by rabbah — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mrutkows) [reply README.md:40]: will remove (cut and pasted too much)

**Pull request merged**